src/swarmpal/toolboxes/tfa/tfa_processor.py

```diff
@@ -15,192 +15,662 @@
 
-    DEFAULTS = {
-        "measurements": [],
-        "model": "'CHAOS-Core' + 'CHAOS-Static'",
-        "auxiliaries": ["QDLat", "QDLon", "MLT"],
-        "sampling_step": None,
-    }
+    def convert_to_mfa(self):
+        """Convert field from NEC to MFA coordinates"""
+        # only run for NEC variables
+        if "NEC" in self.VARNAME:
+            B_MFA = tfalib.mfa(
+                self.xarray["B_NEC"].data, self.xarray["B_NEC_Model"].data
+            )
+
+            # Use the append_array() function when it gets updated
+            # self.append_array("B_MFA", B_MFA, dims=("Timestamp", "MFA"))
+            #
+            # For now, its being done manually
+            self.xarray = self.xarray.assign_coords({"MFA": ["M", "F", "A"]})
+            self.xarray = self.xarray.assign({"B_MFA": (("Timestamp", "MFA"), B_MFA)})
+        else:
+            logging.warn("TfaInput: Cannot run Conversion to MFA for non-NEC inputs")
+
+    def __init__(
+        self, varname="B_NEC", remove_chaos=False, sampling_time="PT1S", *args, **kwargs
+    ):
+        # Select variable and set flag and maximum acceptable flag value
+        self.VARNAME = varname
+        if self.VARNAME == "B_NEC":
+            self.FLAGNAME = "Flags_B"
+            self.MAX_FLAG_VAL = 30
+        elif self.VARNAME == "F":
+            self.FLAGNAME = "Flags_F"
+            self.MAX_FLAG_VAL = 63
+        elif self.VARNAME == "E_NEC":
+            self.FLAGNAME = "Flags_TII"
+            self.MAX_FLAG_VAL = 23
+        elif self.VARNAME == "n":
+            self.FLAGNAME = "Flags_LP"
+            self.MAX_FLAG_VAL = 63
+        elif self.VARNAME == "Bubble_Probability":
+            self.FLAGNAME = "Flags_F"
+            self.MAX_FLAG_VAL = 1000
+        elif self.VARNAME == "FAC":
+            self.FLAGNAME = "Flags_F"
+            self.MAX_FLAG_VAL = 1000
+        elif self.VARNAME == "EEF":
+            self.FLAGNAME = "Flags"
+            self.MAX_FLAG_VAL = 1000
+        elif self.VARNAME == "Eh_XYZ":
+            self.FLAGNAME = "Quality_flags"
+            self.MAX_FLAG_VAL = 0
+        elif self.VARNAME == "Ev_XYZ":
+            self.FLAGNAME = "Quality_flags"
+            self.MAX_FLAG_VAL = 0
+        else:
+            sys.exit("TfaInput: unrecognized 'varname' input: " + varname)
+
+        self.DEFAULTS["measurements"] = [self.VARNAME, self.FLAGNAME]
+
+        # Check if CHAOS field removal has been requested and if yes, add
+        # it to the input parameters so that it will be retrieved
+        if remove_chaos:
+            self.REMOVE_CHAOS = True
+            self.DEFAULTS["model"] = "'CHAOS-Core' + 'CHAOS-Static'"
+        else:
+            self.REMOVE_CHAOS = False
+            self.DEFAULTS["model"] = "None"
+
+        # set pad_times to either 3 hours, if Low-res data), or 5 minutes if
+        # high-res
+        self.COLLECTION = kwargs["collection"]
+        if "HR" in self.COLLECTION:
+            self.DEFAULTS["pad_times"] = [
```

Comment:
Let's find a cleaner way to list out all the defaults - maybe in a separate configuration file